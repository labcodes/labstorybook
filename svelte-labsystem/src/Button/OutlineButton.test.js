import { render } from "@testing-library/svelte";
import OutlineButton from "./OutlineButton.svelte";

describe("OutlineButton", () => {
  it("renders with base props", async () => {
    const { container } = render(OutlineButton, { text: "test" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing disabled as true", async () => {
    const { container, getByText } = render(OutlineButton, {
      text: "test",
      disabled: true,
    });

    expect(container).toMatchSnapshot();
    expect(getByText("test").disabled).toBeTruthy();
  });

  it("renders as expected when passing a light skin", async () => {
    const { container } = render(OutlineButton, {
      text: "test",
      skin: "light",
    });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a small size", async () => {
    const { container } = render(OutlineButton, {
      text: "test",
      size: "small",
    });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a wallet icon", async () => {
    const { container } = render(OutlineButton, {
      text: "test",
      icon: "wallet",
    });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing fullWidth prop", async () => {
    const { container } = render(OutlineButton, {
      text: "test",
      fullWidth: true,
    });
    expect(container).toMatchSnapshot();
  });
});
