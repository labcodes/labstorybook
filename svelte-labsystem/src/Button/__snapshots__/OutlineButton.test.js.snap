// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton renders as expected when passing a light skin 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--normal lab-btn--light"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`OutlineButton renders as expected when passing a small size 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--small"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`OutlineButton renders as expected when passing a wallet icon 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--normal"
    >
      <span
        class="lab-icon lab-icon--wallet lab-icon--white lab-icon--petit lab-btn__icon"
      />
       
      test
    </button>
  </div>
</body>
`;

exports[`OutlineButton renders as expected when passing disabled as true 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--normal"
      disabled=""
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`OutlineButton renders as expected when passing fullWidth prop 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--normal lab-btn--block"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`OutlineButton renders with base props 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--outline lab-btn--normal"
    >
       
      test
    </button>
  </div>
</body>
`;
